packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -141,44 +124,38 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
    */
   private modelDirectiveConfig: Map<string, ModelDirectiveConfiguration> = new Map();
   constructor(options: ModelTransformerOptions = {}) {
     super('amplify-model-transformer', directiveDefinition);
     this.options = this.getOptions(options);
+    const rdsGenerator = new RdsModelResourceGenerator();
+    this.resourceGeneratorMap.set(DDB_DB_TYPE, new DynamoModelResourceGenerator());
+    this.resourceGeneratorMap.set(MYSQL_DB_TYPE, rdsGenerator);
   }
 
   before = (ctx: TransformerBeforeStepContextProvider): void => {
-    // add model related-parameters to the root stack
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBModelTableReadIOPS, {
-      description: 'The number of read IOPS the table should support.',
-      type: 'Number',
-      default: 5,
-    });
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBModelTableWriteIOPS, {
-      description: 'The number of write IOPS the table should support.',
-      type: 'Number',
-      default: 5,
-    });
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBBillingMode, {
-      description: 'Configure @model types to create DynamoDB tables with PAY_PER_REQUEST or PROVISIONED billing modes.',
-      default: 'PAY_PER_REQUEST',
-      allowedValues: ['PAY_PER_REQUEST', 'PROVISIONED'],
-    });
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBEnablePointInTimeRecovery, {
-      description: 'Whether to enable Point in Time Recovery on the table.',
-      type: 'String',
-      default: 'false',
-      allowedValues: ['true', 'false'],
-    });
-    ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBEnableServerSideEncryption, {
-      description: 'Enable server side encryption powered by KMS.',
-      type: 'String',
-      default: 'true',
-      allowedValues: ['true', 'false'],
-    });
+    const datasourceMapValues: Array<DatasourceType> = Array.from(ctx.modelToDatasourceMap.values());
+    if (datasourceMapValues.some((value) => value.dbType === DDB_DB_TYPE && value.provisionDB)) {
+      this.resourceGeneratorMap.get(DDB_DB_TYPE)?.enableGenerator();
+      this.resourceGeneratorMap.get(DDB_DB_TYPE)?.enableProvisioned();
+    }
+    if (datasourceMapValues.some((value) => value.dbType === MYSQL_DB_TYPE && !value.provisionDB)) {
+      this.resourceGeneratorMap.get(MYSQL_DB_TYPE)?.enableGenerator();
+      this.resourceGeneratorMap.get(MYSQL_DB_TYPE)?.enableUnprovisioned();
+    }
+    if (datasourceMapValues.length === 0) {
+      // Just enable DynamoDB provisioned, legacy use
+      this.resourceGeneratorMap.get(DDB_DB_TYPE)?.enableGenerator();
+      this.resourceGeneratorMap.get(DDB_DB_TYPE)?.enableProvisioned();
+    }
+    // We only store this in the model because some of the required override methods need to pass through to the
+    // Resource generators, but do not have access to the context
+    this.modelToDatasourceMap = ctx.modelToDatasourceMap;
```

Comment:
avoid the copy if possible unless you have a strong reason

Reply:
It's completely necessary unless we remove the override methods from the `ModelTransformerBase` interface which we might be able to do per an above comment